frontend/src/diagram/sprotty/views/LabelView.tsx

```diff
@@ -48,12 +49,40 @@ export class LabelView extends SLabelView {
         styleObject['text-decoration'] += ' line-through';
       }
     }
+    if (label.type.includes('center')) {
+      styleObject['text-anchor'] = 'middle';
+    }
     const iconVerticalOffset = -12;
+
+    const text = label.text;
+    const Text = () => {
+      const lines = text.split('\n');
+      if (lines.length == 1) {
+        return text;
+      } else {
+        return lines.map((line, index) => {
+          if (index === 0) {
+            return <tspan x="0">{line}</tspan>;
+          } else {
+            if (line.length == 0) {
+              // avoid tspan to be ignored if there is only a line return
+              line = ' '; //$NON-NLS-1$
+            }
+            return (
+              <tspan x="0" dy={fontSize}>
+                {line}
+              </tspan>
+            );
+          }
+        });
+      }
+    };
```

Comment:
This component can be extracted and given props. Overwise its lifecycle is a bit odd since it is based on a closure